iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/memory/MemoryManager.java

```diff
@@ -294,35 +299,35 @@ public synchronized void releaseWithOutNotify(IMemoryBlock block) {
    *
    * @param name the name of memory manager
    * @param sizeInBytes the total memory size in bytes of memory manager
-   * @param enable whether memory management is enabled
+   * @param enabled whether memory management is enabled
    * @return the memory manager
    */
   public synchronized MemoryManager getOrCreateMemoryManager(
-      String name, long sizeInBytes, boolean enable) {
-    if (sizeInBytes <= 0) {
-      LOGGER.warn("getOrCreateMemoryManager {}: sizeInBytes should be positive", name);
-    }
-    if (this.enable
-        && sizeInBytes + this.allocatedMemorySizeInBytes > this.totalMemorySizeInBytes) {
-      LOGGER.warn(
-          "getOrCreateMemoryManager failed: total memory size {} bytes is less than allocated memory size {} bytes",
-          sizeInBytes,
-          allocatedMemorySizeInBytes);
-      return null;
-    }
+      String name, long sizeInBytes, boolean enabled) {
     return children.compute(
         name,
         (managerName, manager) -> {
-          if (manager != null) {
+          if (sizeInBytes <= 0) {
+            LOGGER.warn("getOrCreateMemoryManager {}: sizeInBytes should be positive", name);
+          }
+          if (this.enabled
+              && sizeInBytes + this.allocatedMemorySizeInBytes > this.totalMemorySizeInBytes) {
             LOGGER.warn(
-                "getOrCreateMemoryManager failed: memory manager {} already exists, it's size is {}, enable is {}",
+                "getOrCreateMemoryManager failed: total memory size {} bytes is less than allocated memory size {} bytes",
+                sizeInBytes,
+                allocatedMemorySizeInBytes);
+            return null;
+          }
```

Comment:
If the manager already exists, is the check correct?

Reply:
Ok